test/portfolios/PortfolioUpgradeable.test.ts

```diff
@@ -7,27 +7,7 @@ export function testPortfolioUpgradeable() {
       const addr_before_upgrade = await upgrades.erc1967.getImplementationAddress(this.portfolio.address)
 
       const NewPortfolio = await ethers.getContractFactory(this.upgradeTo, this.owner)
-      const newPortfolio = await upgrades.upgradeProxy(this.portfolio.address, NewPortfolio, {
```

Comment:
hmmmm, I think in the future we might have contracts with 'reinitializer' and we want to check that after calling it, our contract is not messed up. Although that 'reinitializer' function will take Strategy/Version dependent args.
Any easy way to generalize the 'reinitializer' logic in our base classes, so we can test it? If it takes a long time to implement, it is not important.

Reply:
reinitialize function takes different arguments and different reinitializer modifier per contract and its version. For example, TraderJoeV2 takes LBRouter address and LBPool address while TraderJoeV3 takes something else. It can't be generalized to be included in our base contracts.

Reply:
yeah, I was thinking about this... It would be nice if the testStrategy could not just take-in the functions that it needs to execute, but also take in a 'Config object' when it is calle from higher level tests like in DnsVectorStrategy.test.ts

testStrategy("Dns Vector Strategy - Deploy", deployDnsStrategy, "RoleableStrategyV2", [
  testAum,
  testCollaterizationAndDeltaNeutrality,
  testRebalanceSafetyLimits,
],
**{blablabla: "asdfasf"}**)

so the dependent tests could be configured somehow

just an idea though that you can think about

Reply:
Generalizing update config so that upgradeStrategy function can handle both upgrade and upgrade and call is already done. Will be included in the following PR.